### ISSUE TITLE
MetricTracker best_metric() returns the step instead of the best value

### ISSUE THREAD

**Issue by bruce-willis** [Lightning-AI/metrics#1304]
## 🐛 Bug

`MetricTracker.best_metric()` function returns the **step** with the highest metric value instead of the best **value** itself.

### To Reproduce
Please see [colab] for a full example and proposed solution.

#### Code sample
```py
from torchmetrics import Accuracy, MetricTracker
import torch
_ = torch.manual_seed(42)
tracker = MetricTracker(Accuracy(num_classes=10))
for epoch in range(5):
    tracker.increment()
    for batch_idx in range(5):
        preds, target = torch.randint(10, (100,)), torch.randint(10, (100,))
        tracker.update(preds, target)
best_acc, which_epoch = tracker.best_metric(return_step=True)

best_acc2 = tracker.best_metric(return_step=False)

assert best_acc == best_acc2 # does not hold but should
assert which_epoch == best_acc2 # instead this condition holds but shouldn't
```

### Expected behavior
Function `best_metric` without any parameters should return the best value:
> Returns the highest metric out of all tracked.  
> [best_metric function docstring](https://github.com/Lightning-AI/metrics/blob/604ed8061449a88e07ccbf764237fd47200c2038/src/torchmetrics/wrappers/tracker.py#L163)

### Environment
Does not depend on the version.

### Inconsistent behavior even with `return_step=True` 
In fact, even with `return_step=True` the order is swapped, the function should return:
```py
return idx.item(), best.item()
```
index first and best value second ([source code](https://github.com/Lightning-AI/metrics/blob/604ed8061449a88e07ccbf764237fd47200c2038/src/torchmetrics/wrappers/tracker.py#L176)), but in current version this function return best value first and index second (`best_acc, which_epoch = ...` from example).

### Proposed solution
Use a namedtuple (values, indices) ([pytorch max function doc](https://pytorch.org/docs/stable/generated/torch.max.html)) instead of implicit unpacking.  

Replace the line with the error
```py
idx[k], best[k] = out[0].item(), out[1].item()
```
from [torchmetrics source code](https://github.com/Lightning-AI/metrics/blob/604ed8061449a88e07ccbf764237fd47200c2038/src/torchmetrics/wrappers/tracker.py#L197)
with:
```py
idx[k], best[k] = out.indices.item(), out.values.item()
```

It would be more readable and easier to understand and more prone to error.

If you agree with the proposal I can submit a follow-up pull request.

### Proposed test case
Add the following test case:
```py
# Assert that best_metric returns both index and value
val, idx = tracker.best_metric(return_step=True)
# Assert that best_metric returns just value without any parameters
val2 = tracker.best_metric()
assert val == val2
```
to the current [test](https://github.com/Lightning-AI/metrics/blob/604ed8061449a88e07ccbf764237fd47200c2038/tests/unittests/wrappers/test_tracker.py#L120).

### Backward compatibility
I think in order not to break backward compatibility this function should return the best value first and index second (as it shown in current examples). And probably the order should be placed in the documentation.

[colab]: https://colab.research.google.com/drive/1LauxXyQZTkchhPuc5vD2zdOu9sP7UZX-?usp=sharing
